### PR TITLE
nix: add nix-team to CODEOWNERS

### PR DESCRIPTION
**Pull request by Mic92**
## Description of changes

Depends on https://github.com/NixOS/org/issues/24

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -75,8 +75,8 @@
 # @raitobezarius is not "code owner", but is listed here to be notified of changes
 # pertaining to the Nix package manager.
 # i.e. no authority over those files.
-pkgs/tools/package-management/nix/                    @raitobezarius
-nixos/modules/installer/tools/nix-fallback-paths.nix  @raitobezarius
+pkgs/tools/package-management/nix/                    @raitobezarius @NixOS/nix-team
```
Comment (Mic92) [inline .github/CODEOWNERS:78]: Mhm. Do teams not work on this level? We also got @NixOS/security up there.

Comment (Mic92) [reply .github/CODEOWNERS:78]: nixpkgs needs to be added to the nix-team.

**Comment by Mic92** [comment]
cc @roberth

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -75,8 +75,8 @@
...
+nixos/modules/installer/tools/nix-fallback-paths.nix  @raitobezarius @NixOS/nix-team
```
Comment (roberth) [inline .github/CODEOWNERS:79]: Considering the comment above,

```suggestion
pkgs/tools/package-management/nix/                    @NixOS/nix-team @raitobezarius 
nixos/modules/installer/tools/nix-fallback-paths.nix  @NixOS/nix-team @raitobezarius 
```

**Review by roberth — APPROVED** [review]
Note the suggestion, but :+1:

**Comment by Mic92** [comment]
Backport https://github.com/NixOS/nixpkgs/pull/336734

**Pull request merged**